### DISCUSSION TITLE
what should i do, if i want to scan somekeys in a specific node?

### DISCUSSION THREAD

**Discussion by NaraLuwan** [redis/go-redis#1856] [Ideas]
now, the scan fun only the following params are supported:
Scan(ctx context.Context, cursor uint64, match string, count int64)

Is it possible to add the node param? If not, how do we do that now?

**Comment by vmihailenco** [comment]
You could use `ForEachNode` to iterate over all nodes and pick the one you need. But we should probably add something like `cluster.ShardForKey(string) *redis.Client` to get the node for the key.

**Reply by NaraLuwan** [reply]
but my redis has a proxy cluster, When I create the client, I call method NewClient instead of method NewClusterClient

**Comment by NaraLuwan** [comment]
I tried it this way, it works for me

	ScanNode := func(ctx context.Context, rdb *redis.Client, cursor uint64, nodeId string) *redis.StringSliceCmd {
		cmd := redis.NewStringSliceCmd(ctx, "scan", cursor, nodeId)
		_ = rdb.Process(ctx, cmd)
		return cmd
	}

	res, err := ScanNode(ctx, rdb, 0, "xxx").Result()
